### PR TITLE
Asset video sidebar

### PR DESCRIPTION
**Pull request by lukmzig**
## Changes in this pull request
Part of https://github.com/pimcore/studio-ui-bundle/issues/80

## Additional info
- add endpoint download video by thumbnail definition
- add endpoint to stream video by thumbnail definition
- add endpoint to stream video image thumbnail 
- add possibility to update custom settings

### CODE REVIEW

File: src/Exception/InvalidThumbnailConfigurationException.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+ * @internal
+ */
+final class InvalidThumbnailConfigurationException extends AbstractApiException
+{
+    public function __construct(string $type)
+    {
+        parent::__construct(
+            400,
```
Comment (martineiber) [inline src/Exception/InvalidThumbnailConfigurationException.php]: We can use the HttpCode Enum here.

File: src/Exception/InvalidThumbnailException.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+/**
+ * @internal
+ */
+final class InvalidThumbnailException extends AbstractApiException
+{
+    public function __construct(string $thumbnailName)
+    {
+        parent::__construct(
+            400,
```
Comment (martineiber) [inline src/Exception/InvalidThumbnailException.php]: Same thing... Check all the other places too, there are quite a few.

File: src/Util/Traits/StreamedResponseTrait.php [hunk 3]
```diff
@@ -16,18 +16,24 @@
...
  * @internal
  */
 trait StreamedResponseTrait
 {
+    /**
+     * @throws ElementStreamResourceNotFoundException
+     */
     protected function getStreamedResponse(
-        Asset $element,
+        Asset|Video\ImageThumbnailInterface $element,
```
Comment (martineiber) [inline src/Util/Traits/StreamedResponseTrait.php]: Can we move the Video part to the use section and only have the ImageThumbnailInterface here?

**Review by martineiber — CHANGES_REQUESTED** [review]
Some minor changes.

I'm missing some Unit Tests for this ;)

**Review by martineiber — APPROVED** [review]
LGTM

**Comment by lukmzig** [comment]
@martineiber I implemented the feedback from your review. As we discussed internally, we for now not provide this unit tests

**Pull request merged**